File: ruby/rbfind/lib/rbfind/finderror.rb
# frozen_string_literal: true

module RbFind

  INVALID_RANGE_MINDEPTH_MAXDEPTH = 'Invalid range for mindepth and maxdepth'
  INVALID_RANGE_MINLASTMOD_MAXLASTMOD = 'Invalid range for minlastmod and maxlastmod'
  INVALID_RANGE_MINSIZE_MAXSIZE = 'Invalid range for minsize and maxsize'
  STARTPATH_NOT_DEFINED = 'Startpath not defined'
  STARTPATH_NOT_FOUND = 'Startpath not found'
  STARTPATH_NOT_MATCH_SETTINGS = 'Startpath does not match find settings'
  STARTPATH_NOT_PATHNAME = 'Startpath not a Pathname instance'
  STARTPATH_NOT_READABLE = 'Startpath not readable'

  # FindError - custom exception class
  class FindError < StandardError
    def initialize(msg = 'FindError occurred')
      super(msg)
    end
  end
end

File: elixir/exfind/lib/finderror.ex
defmodule ExFind.FindError do
  @moduledoc """
  Documentation for `ExFind.FindError`.
  """

  defexception message: "An unknown error occurred."

  @startpath_not_defined "Startpath not defined"

  @startpath_not_found "Startpath not found"

  @startpath_not_readable "Startpath not readable"

  @startpath_not_match_find_settings "Startpath does not match find settings"

  @invalid_range_mindepth_and_maxdepth "Invalid range for mindepth and maxdepth"

  @invalid_range_minsize_and_maxsize "Invalid range for minsize and maxsize"

  @invalid_range_minlastmod_and_maxlastmod "Invalid range for minlastmod and maxlastmod"

  def startpath_not_defined, do: @startpath_not_defined

  def startpath_not_found, do: @startpath_not_found

  def startpath_not_readable, do: @startpath_not_readable

  def startpath_not_match_find_settings, do: @startpath_not_match_find_settings

  def invalid_range_mindepth_and_maxdepth, do: @invalid_range_mindepth_and_maxdepth

  def invalid_range_minsize_and_maxsize, do: @invalid_range_minsize_and_maxsize

  def invalid_range_minlastmod_and_maxlastmod, do: @invalid_range_minlastmod_and_maxlastmod

end
